aspnetcore/fundamentals/host/platform-specific-configuration.md

```diff
@@ -0,0 +1,451 @@
+
+```console
+dotnet nuget locals all --clear
+```
+
+**Activation from a class library**
+
+1. Compile the *HostingStartupLibrary* class library with the [dotnet build](/dotnet/core/tools/dotnet-build) command.
+1. Add the class library's assembly name of *HostingStartupLibrary* to the `ASPNETCORE_HOSTINGSTARTUPASSEMBLIES` environment variable.
+1. *bin*-deploy the class library's assembly to the app by copying the *HostingStartupLibrary.dll* file from the class library's compiled output to the app's *bin* folder.
+1. Compile and run the app. An `<ItemGroup>` in the app's project file references the class library's assembly (*.\bin\Debug\netcoreapp2.1\HostingStartupLibrary.dll*) (a compile-time reference):
+
+   ```xml
+   <ItemGroup>
+     <Reference Include=".\bin\Debug\netcoreapp2.1\HostingStartupLibrary.dll">
+       <HintPath>.\bin\Debug\netcoreapp2.1\HostingStartupLibrary.dll</HintPath>
+       <SpecificVersion>False</SpecificVersion>
+     </Reference>
+   </ItemGroup>
+   ```
+1. Observe that the service configuration key values rendered by the Index page match the values set by the class library's `ServiceKeyInjection.Configure` method.
+
+**Activation from a runtime store-deployed assembly**
+
+1. The *StartupDiagnostics* project uses [PowerShell](/powershell/scripting/powershell-scripting) to modify its *StartupDiagnostics.deps.json* file. PowerShell is installed by default on Windows OS starting with Windows 7 SP1 and Windows Server 2008 R2 SP1. To obtain PowerShell on other platforms, see [Installing Windows PowerShell](/powershell/scripting/setup/installing-windows-powershell).
+1. Build the *StartupDiagnostics* project. After the project is built, a build target in the project file automatically:
+   * Triggers the PowerShell script to modify the *StartupDiagnostics.deps.json* file.
+   * Moves the *StartupDiagnostics.deps.json* file to the user profile's *additionalDeps* folder.
+1. Execute the `dotnet store` command at a command prompt in the hosting startup's directory to store the assembly and its dependencies in the user profile's runtime store:
+
+   ```console
+   dotnet store --manifest StartupDiagnostics.csproj --runtime win7-x64
+   ```
+   For Windows, the command uses the `win7-x64` runtime identifier (RID). When providing the hosting startup for a different runtime, substitute the correct RID.
+1. Set the environment variables:
+   * Add the assembly name of *StartupDiagnostics* to the `ASPNETCORE_HOSTINGSTARTUPASSEMBLIES` environment variable.
+   * On Windows, set the `DOTNET_ADDITIONAL_DEPS` environment variable to `%UserProfile%\.dotnet\x64\additionalDeps\StartupDiagnostics\`. On macOS/Linux, set the `DOTNET_ADDITIONAL_DEPS` environment variable to `/Users/<USER>/.dotnet/x64/additionalDeps/StartupDiagnostics/`, where `<USER>` is the user profile that contains the hosting startup.
```

Comment:
I fixed the command by showing `<RID>`. Therefore, there's only one sentence left specific to non-Windows ... this one. Let's see on the next commit. It might not be worth it to copy the entire set of steps into three tabs only to accommodate the single sentence.